plasmapy/formulary/frequencies.py

```diff
@@ -50,29 +57,25 @@ def gyrofrequency(B: u.T, particle: ParticleLike, signed=False, Z=None) -> u.rad
     B : `~astropy.units.Quantity`
         The magnetic field magnitude in units convertible to tesla.
 
-    particle : `~plasmapy.particles.particle_class.Particle`
-        Representation of the particle species (e.g., 'p' for protons, 'D+'
-        for deuterium, or 'He-4 +1' for singly ionized helium-4). If no
-        charge state information is provided, then the particles are assumed
-        to be singly charged.
-
-    signed : `bool`, optional
-        The gyrofrequency can be defined as signed (negative for electron,
-        positive for ion). Default is `False` (unsigned, i.e. always
-        positive).
-
-    Z : `float` or `~astropy.units.Quantity`, optional
-        The average ionization (arithmetic mean) for a plasma where
-        a macroscopic description is valid. If this quantity is not
-        given then the charge number of the ion
-        is used. This is effectively an average gyrofrequency for the
-        plasma where multiple charge states are present, and should
-        not be interpreted as the gyrofrequency for any single particle.
-        If not provided, it defaults to the charge number of the ``particle``.
+    particle : |particle-like|
+        Representation of the particle species (e.g., ``'p'`` for
+        protons, ``'D+'`` for deuterium, or ``'He-4 +1'`` for singly
+        ionized helium-4).
+
+    signed : `bool`, default: `False`
+        The gyrofrequency can be defined as signed, where its sign is
+        the sign of the |charge number|. Defaults to unsigned (i.e.,
+        always positive).
+
+    Z : real number or `~astropy.units.Quantity`, optional
```

Comment:
Not clear to me from the docs if Z and mass_numb override the particle if both are provided?

Reply:
This is a good point.  I just created #2024, since your suggestion is related to the broader issue of standardizing user experience for particles.  In the meantime, I'll make an update to the parameter description here.

Reply:
I forgot to mention that if either `Z` or `mass_numb` is provided in both `particle` and as a separate argument, then `@particle_input` will raise an exception.